modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/new-js/components/field-sets/FieldSetModal.js

```diff
@@ -0,0 +1,472 @@
+
+import getFieldsWithoutOptions from '../../../js/components/field-sets/actions/getFieldsWithoutOptions.es';
+import {usePropagateFieldSet} from '../../../js/components/field-sets/actions/usePropagateFieldSet.es';
+import TranslationManager from '../../../js/components/translation-manager/TranslationManager.es';
+import {addItem, updateItem} from '../../../js/utils/client.es';
+import {getFieldSetDDMForm} from '../../../js/utils/dataConverter.es';
+import {isDataLayoutEmpty} from '../../../js/utils/dataLayoutVisitor.es';
+import {normalizeDataLayout} from '../../../js/utils/normalizers.es';
+import {errorToast, successToast} from '../../../js/utils/toast.es';
+import {FormBuilder} from '../../FormBuilder';
+import {EVENT_TYPES} from '../../eventTypes';
+import sidebarReducer from '../../reducers/sidebarReducer';
+
+const DEFAULT_DATA_DEFINITION = {
+	dataDefinitionFields: [],
+};
+
+const DEFAULT_DATA_LAYOUT = {
+	dataLayoutFields: {},
+	dataLayoutPages: [],
+	dataRules: [],
+	paginationMode: 'single-page',
+};
+
+const ModalContent = ({
+	fieldSet: fieldSetProp,
+	onClose,
+	onUpdate,
+	updateFieldSetList,
+}) => {
+	const dispatch = useForm();
+	const {contentType} = useConfig();
+	const {
+		dataDefinitionId,
+		defaultLanguageId,
+		editingLanguageId,
+	} = useFormState();
+	const {dataDefinition, dataLayout} = useFormState({
+		schema: ['dataDefinition', 'dataLayout'],
+	});
+
+	const [name, setName] = useState(dataDefinition.name ?? {});
+	const changeEditingLanguageId = (editingLanguageId) =>
+		dispatch({
+			payload: {editingLanguageId},
+			type: EVENT_TYPES_CORE.LANGUAGE.CHANGE,
+		});
+
+	const onSave = async () => {
+		const {
+			availableLanguageIds,
+			dataDefinitionFields,
+		} = dataDefinition.build();
+
+		const {dataLayoutPages} = dataLayout.build();
+
+		try {
+			if (!name[defaultLanguageId]) {
+				changeEditingLanguageId(defaultLanguageId);
+				throw new Error(
+					Liferay.Language.get('please-enter-a-valid-title')
+				);
+			}
+
+			const fieldsWithoutOptions = getFieldsWithoutOptions(
+				dataDefinitionFields,
+				defaultLanguageId
+			);
+
+			if (fieldsWithoutOptions.length) {
+				throw new Error(
+					Liferay.Util.sub(
+						Liferay.Language.get(
+							'at-least-one-option-should-be-set-for-field-x'
+						),
+						fieldsWithoutOptions[0].label[defaultLanguageId]
+					)
+				);
+			}
+
+			if (dataDefinitionId) {
+
+				// update fieldSet
+
+				const updatedFieldSet = {
+					...fieldSetProp,
+					availableLanguageIds,
+					dataDefinitionFields,
+					defaultDataLayout: normalizeDataLayout(
+						{
+							...fieldSetProp.defaultDataLayout,
+
+							dataLayoutPages: dataLayoutPages.map((page) => {
+								return {
+									...page,
+									description: {
+										[defaultLanguageId]: '',
+									},
+
+									title: {
+										[defaultLanguageId]: '',
+									},
+								};
+							}),
+
+							name,
+						},
+						defaultLanguageId
+					),
```

Comment:
Just a doubt why do we need to do `normalizeDataLayout`? In theory, the data for `defaultDataLayout` is the same as `dataLayout`.

Reply:
We are going to check it again!